src/api/block.js

```diff
@@ -8,7 +8,11 @@ module.exports = (send) => {
     stat: argCommand(send, 'block/stat'),
     put (file, cb) {
       if (Array.isArray(file)) {
-        return cb(null, new Error('block.put() only accepts 1 file'))
+        let err = new Error('block.put() only accepts 1 file')
+        if (typeof cb !== 'function') {
```

Comment:
please also check for `Promise` being defined: https://github.com/ipfs/js-ipfs-api/blob/master/src/api/log.js#L8